src/utilities/aerodynamics/aerodyn_inflow.hpp

```diff
@@ -614,24 +747,26 @@ class AeroDynInflowLibrary {
             // Create new turbine data to store the updates
             // Note: TurbineData and MeshData are validated during construction, so no need to
             // perform additional checks here
-            turbines_.emplace_back(tc);
-            auto& td = turbines_.back();
+            turbines.emplace_back(tc);
+            auto& td = turbines.back();
 
             // Call setup rotor for each turbine
             ADI_C_SetupRotor(
                 &turbine_number,                            // input: current turbine number
                 &is_horizontal_axis_int,                    // input: 1: HAWT, 0: VAWT or cross-flow
                 tc.reference_position.data(),               // input: turbine reference position
                 td.hub.position.data()->data(),             // input: initial hub position
-                td.hub.orientation.data()->data(),          // input: initial hub orientation
+                td.hub.orientation.data()->data()->data(),  // input: initial hub orientation
                 td.nacelle.position.data()->data(),         // input: initial nacelle position
-                td.nacelle.orientation.data()->data(),      // input: initial nacelle orientation
-                &td.n_blades,                               // input: number of blades
-                td.blade_roots.position.data()->data(),     // input: initial blade root positions
-                td.blade_roots.orientation.data()->data(),  // input: initial blade root orientation
-                &td.blade_nodes.n_mesh_points,              // input: number of mesh points
-                td.blade_nodes.position.data()->data(),     // input: initial node positions
-                td.blade_nodes.orientation.data()->data(),  // input: initial node orientation
+                td.nacelle.orientation.data()->data()->data(),  // input: initial nacelle orientation
+                &td.n_blades,                                   // input: number of blades
+                td.blade_roots.position.data()->data(),  // input: initial blade root positions
+                td.blade_roots.orientation.data()->data()->data(
+                ),                                       // input: initial blade root orientation
+                &td.blade_nodes.n_points,                // input: number of mesh points
+                td.blade_nodes.position.data()->data(),  // input: initial node positions
+                td.blade_nodes.orientation.data()->data()->data(
```

Comment:
For these nested `->data()` calls, it might be better to use the `[]` operators.  i.e. instead of `orientation.data()->data()->data()` write `orientation[0][0].data()`.  

In real terms, it might be better for us to use a single, flat block of memory and write a simple class or two with an overloaded `operator()` and `.data()` for these data structures.  We can look into this in the future.

Reply:
I think using `[0][0].data()` makes more sense, I just figured it using `data()` for each layer was more in line with the C++ way of doing things. We certainly could make a custom class, but I think the vector of arrays express the intent very clearly.